### DISCUSSION TITLE
Linux router netflow

### DISCUSSION THREAD

**Discussion by kwbrd** [akvorado/akvorado#102] [Q&A]
Hello there!

Wanna try to collect netflow from some deb-linux vpn-gateways.
Anybody did so?
Will snmpd works correct?
Or any other underlying potential problems?

**Comment by kornix** [comment]
Hi.

Since netflow generation depends on distro (e. g. there is netflow module for iptables in older distros, but no such production grade module for nftables deployed in newer ones), i would suggest using hsflowd (https://sflow.net/downloads.php) for sflow generation. And yes, as i can see, it works correct with snmpd.

**Reply by eahlys** [reply]
I just did this however hsflowd will report sflows containing one unknown interface (because traffic isn't flowing though my debian box, it originates/terminates here as it's an host and not really a router per se). Those flows are then discarded by akvorado because it cannot gather interfaces informations using snmp (it will work for one interface, but the other does not exist as there are no other interface)
Did you have any luck with this ?

Thanks !

**Reply by kornix** [reply]
1. Those flows has not to be discarded. Check 'interface-classifiers' section of config.
2. I have separate VLANs for different segments of the network, so they are presented as separate interfaces to hsflowd.
